Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -179,63 +179,105 @@ public class PublishPacket: CStruct {
     func withCStruct<Result>(_ body: (aws_mqtt5_packet_publish_view) -> Result) -> Result {
         var raw_publish_view = aws_mqtt5_packet_publish_view()
 
-        raw_publish_view.qos = aws_mqtt5_qos(UInt32(qos.rawValue))
+        raw_publish_view.qos = self.qos.nativeValue
         raw_publish_view.retain = retain
         return topic.withByteCursor { topicCustor in
             raw_publish_view.topic =  topicCustor
             return withOptionalAWSByteCursorFromData(to: payload) { cByteCursor in
                 raw_publish_view.payload = cByteCursor
 
                 let _payloadFormatIndicatorInt: aws_mqtt5_payload_format_indicator? =
-                                                                    payloadFormatIndicator?.rawValue ?? nil
+                                                                    payloadFormatIndicator?.nativeValue ?? nil
                 let _messageExpiryInterval: UInt32? = try? messageExpiryInterval?.secondUInt32() ?? nil
 
                 return withOptionalUnsafePointers(
                     _payloadFormatIndicatorInt,
                     topicAlias,
                     _messageExpiryInterval) { payloadPointer, topicAliasPointer, messageExpiryIntervalPointer in
-                    if let _payloadPointer = payloadPointer {
-                        raw_publish_view.payload_format = _payloadPointer
-                    }
-
-                    if let _messageExpiryIntervalPointer = messageExpiryIntervalPointer {
-                        raw_publish_view.message_expiry_interval_seconds = _messageExpiryIntervalPointer
-                    }
-
-                    if let _topicAliasPointer = topicAliasPointer {
-                        raw_publish_view.topic_alias = _topicAliasPointer
-                    }
-
+                    raw_publish_view.payload_format = payloadPointer
+                    raw_publish_view.message_expiry_interval_seconds = messageExpiryIntervalPointer
+                    raw_publish_view.topic_alias = topicAliasPointer
+                        
                     return withOptionalArrayRawPointer(of: subscriptionIdentifiers) { subscriptionPointer in
 
-                        if let _subscriptionPointer = subscriptionPointer {
-                            raw_publish_view.subscription_identifiers = _subscriptionPointer
+                        if let subscriptionPointer = subscriptionPointer {
```

Comment:
Better to just do `if let subscriptionPointer, subscriptionCount {
}` instead of force unwrap.